tests/common/build.rs

```diff
@@ -47,3 +47,23 @@ pub fn build_with_options(
         modes,
     )
 }
+
+pub fn build_with_options_and_modes(
+    component_dir: &Path,
+    env_name: &str,
+    home: &Path,
+    build_opts: &lal::BuildOptions,
+    modes: lal::ShellModes,
```

Comment:
I'm not sure what to do about this one.

I feel that, for consistency, this should be `&lal::ShellModes`.

But the change required to do this is
```
--- a/src/build.rs
+++ b/src/build.rs
@@ -73,9 +73,9 @@ pub fn build(
     manifest: &Manifest,
     opts: &BuildOptions,
     envname: String,
-    _modes: ShellModes,
+    modes: &ShellModes,
 ) -> LalResult<()> {
-    let mut modes = _modes;
+    let mut modes = modes.clone();
 
     // have a better warning on first file-io operation
     // if nfs mounts and stuff cause issues this usually catches it
```

Which is a bit more invasive than the scope of this PR.
I might have to queue this up in a different PR.